concent_api/concent_api/settings/base.py

```diff
@@ -36,6 +36,7 @@
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'concent_api.middleware.GolemMessagesVersionMiddleware',
     'concent_api.middleware.ConcentVersionMiddleware',
+    'concent_api.middleware.HandleServerErrorMiddleware',
```

Comment:
Middleware is executed from top to bottom. I think it would be test to put ours on top so that it can catch all exceptions that escape, not only those from the view.

Well, not sure about `SecurityMiddleware`. Maybe it's better if ours is below it. All the HTTP 5xx responses should be in JSON but not at the cost of exposing a potential hole.

Please test it to make sure it really works that way - for example this middleware should be able to catch an exception raised by `GolemMessagesVersionMiddleware`.

Reply:
when returning from view, middleware is executed in reversed order. If the middleware itself raises and nothing above it catches it, exception will be automatically converted to HttpResponse thanks to django `convert_exception_to_response` mechanism that kicks in during loading of the middleware

Reply:
Exactly, so if you put in on top, it would be executed as last one when processing response.

Reply:
done